common/utils/index.js

```diff
@@ -612,4 +615,117 @@ function getPlatformManager(context) {
+      description: plan.description,
+      free: plan.free ? true : service.free ? true : false,
+      bindable: plan.bindable ? plan.bindable : service.bindable ? service.bindable : false,
+      planUpdatable: plan.bindable ? true : false,
+      templates: plan.templates ? plan.templates : []
+    }
+  };
+  if (plan.metadata) {
+    planCRD.spec.metadata = plan.metadata;
+  }
+  if (plan.manager) {
+    planCRD.spec.manager = plan.manager;
+  }
+  if (plan.context) {
+    planCRD.spec.context = plan.context;
+  }
+  if (plan.actions) {
+    planCRD.spec.actions = plan.actions;
+  }
+  if (plan.async_ops_supporting_parallel_sync_ops) {
+    planCRD.spec.asyncOpsSupportingParallelSyncOps = plan.async_ops_supporting_parallel_sync_ops;
+  }
+  return planCRD;
+}
+
+function getServiceCrdFromConfig(service) {
+  assert.ok(service.name, 'service.name is required to generate plan crd');
+  assert.ok(service.id, 'service.id is required to generate plan crd');
+  assert.ok(service.description, 'service.description is required to generate plan crd');
+  assert.ok(service.bindable, 'service.bindable is required to generate plan crd');
+
+  let serviceCRD = {
+    apiVersion: 'osb.servicefabrik.io/v1alpha1',
+    kind: 'SFService',
+    metadata: {
+      name: service.id,
+      labels: {
+        'controller-tools.k8s.io': '1.0',
+        serviceId: service.id
+      }
+    },
+    spec: {
+      name: service.name,
+      id: service.id,
+      bindable: service.bindable,
+      description: service.description
+    }
+  };
+  if (service.metadata) {
+    serviceCRD.spec.metadata = service.metadata;
+  }
+  if (service.tags) {
+    serviceCRD.spec.tags = service.tags;
+  }
+  if (service.dashboard_client) {
+    serviceCRD.spec.dashboardClient = service.dashboard_client;
+  }
+  if (service.plan_updateable) {
+    serviceCRD.spec.planUpdateable = service.plan_updateable;
+  }
+  if (service.supported_platform) {
+    serviceCRD.spec.supportedPlatform = service.supported_platform;
+  }
+  if (service.actions) {
+    serviceCRD.spec.actions = service.actions;
+  }
+  if (service.backup_interval) {
+    serviceCRD.spec.backupInterval = service.backup_interval;
+  }
+  if (service.pitr) {
+    serviceCRD.spec.pitr = service.pitr;
+  }
+  return serviceCRD;
+}
+
+function pushServicePlanToApiServer() {
+  if (!config.apiserver.isServiceDefinitionAvailableOnApiserver) {
+    if (!config.apiserver.isServiceDefinitionAvailableOnApiserver) {
+      const eventmesh = require('../../data-access-layer/eventmesh');
+      let promiseList = [];
+      _.each(config.services, service => {
+        const servicePromise = eventmesh.apiServerClient.createOrUpdateServicePlan(getServiceCrdFromConfig(service));
+        promiseList.push(servicePromise);
+        _.each(service.plans, plan => {
+          const planPromise = eventmesh.apiServerClient.createOrUpdateServicePlan(getPlanCrdFromConfig(plan, service));
+          promiseList.push(planPromise);
```

Comment:
use [Promise.map](http://bluebirdjs.com/docs/api/promise.map.html) here. It exactly does what you are trying to do in the above.

Reply:
Done